### ISSUE TITLE
Heroku deployment failed on step 1 when doing a GIT PUSH ORIGIN <BRANCH NAME>

### ISSUE THREAD

**Issue by glasneed** [ga-wdi-boston/full-stack-project#810]
This doesn't look promising. No idea what the issue is.

What I've done so far:
1) Successfully deployed the client side logic. It was on a similar named branch called 'auth' that I merged into master and then did a grunt deploy with no issues.

2) This is my first attempt at starting the back end deploy and my first step was to do a similar git merge for my server side folder. 

It choked below and I've no clue what the issue is. None of the hints below offer anything that I'm comfortable with doing myself ..

``` conor@LIBP45P-10984DY ~/wdi/projects/chores-server (auth) $ gs
On branch auth
nothing to commit, working tree clean
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (auth) $ git push origin auth
To https://github.com/conorjennings/chores.git
 ! [rejected]        auth -> auth (fetch first)
error: failed to push some refs to 'https://github.com/conorjennings/chores.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (auth) $ 
```

**Comment by jordanallain** [comment]
its saying that you have work on the `auth` branch that is on github that you don't have on your local `auth` branch. you should be able to compare the branches by running `git diff remotes/origin/auth` from your terminal when you are on the auth branch.

**Comment by glasneed** [comment]
Well I haven't checked in anything to github yet on my server side. This was my first attempt at doing a deploy. I ran that command but it errored as well 

``` conor@LIBP45P-10984DY ~/wdi/projects/chores-server (auth) $ git diff remotes/origin/auth
fatal: ambiguous argument 'remotes/origin/auth': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'```

**Comment by glasneed** [comment]
I merged my one 'auth' branch into master and then tried to do the ``` git push heroku master  ```command. 

It now gave this error. No idea what "sleepy-taiga" is? My project is simply called "chores". I ran a git remote -v command and results below. This deployment is dead in the water at this point.

``` 
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ git push heroku master
remote: !	No such app as sleepy-taiga-13666.
fatal: repository 'https://git.heroku.com/sleepy-taiga-13666.git/' not found
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ git remote -v
heroku	https://git.heroku.com/sleepy-taiga-13666.git (fetch)
heroku	https://git.heroku.com/sleepy-taiga-13666.git (push)
origin	https://github.com/conorjennings/chores.git (fetch)
origin	https://github.com/conorjennings/chores.git (push)
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ 
```

**Comment by glasneed** [comment]
and the name of the heroku URL in settings tab on https://dashboard.heroku.com is:

```
https://git.heroku.com/kids-chores-nh.git
```
which differs from what GIT is complaining about above.

**Comment by danman01** [comment]
Did your heroku app get deleted, or did you ever run `heroku create`? 

If you need to setup a new app on heroku, I would go through this [setup guide for heroku](https://devcenter.heroku.com/articles/getting-started-with-rails5), _in a separate new app directory_ so you get a hang of it and understand what is going on. Then, in your project's api directory, do the `heroku create` step and beyond. You may first have to delete your existing heroku remotes.

**Comment by glasneed** [comment]
All I did with heroku was follow the "_Getting Set Up_" steps in the Heroku document last week to set up an account. No more than that and haven't been back in since. I did run ``` heroku create ``` to answer your question.

As for deleting heroku remotes and creating a new app in a seperate directory, I think I'll hold off as I don't have sufficient knowledge of this stuff to do this by myself. I thought I just had to follow the deployment steps in the document tonight to wrap up this project. 

At this point, I'll have little choice but to present using localhost tomorrow.

**Comment by jordanallain** [comment]
if you look at your github repo you can select the auth branch

https://github.com/conorjennings/chores/tree/auth

so you did push it at some point. that sleepy-taiga url was the original one heroku created for you when we created the app on heroku

**Comment by BenGitsCode** [comment]
What's the name of the remote in your command line? Is it `https://git.heroku.com/kids-chores-nh.git`?

**Comment by glasneed** [comment]
I took Danny's suggestion and deleted those sleep-taiga URLs as I've no idea what they were. Once I did that a few minutes ago, I was able to then do a ``` git push heroku master `` and it seemed to work ...but its now come up with another mysterious name, dry-hollows. I'll continue on for now but not overly confident in what I'm doing here...


``` 
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ git push heroku master 
Counting objects: 197, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (180/180), done.
Writing objects: 100% (197/197), 60.21 KiB | 0 bytes/s, done.
Total 197 (delta 58), reused 0 (delta 0)
remote: Compressing source files... done.
remote: Building source:
remote: 
remote: -----> Ruby app detected
remote: -----> Compiling Ruby/Rails
remote: -----> Using Ruby version: ruby-2.3.1
remote: -----> Installing dependencies using bundler 1.13.7
remote:        Running: bundle install --without development:test --path vendor/bundle --binstubs vendor/bundle/bin -j4 --deployment
remote:        Fetching gem metadata from https://rubygems.org/.........
remote:        Fetching version metadata from https://rubygems.org/..
remote:        Fetching dependency metadata from https://rubygems.org/.
remote:        Installing i18n 0.7.0
remote:        Installing rake 12.0.0
remote:        Installing concurrent-ruby 1.0.4
remote:        Installing minitest 5.10.1
remote:        Installing thread_safe 0.3.5
remote:        Installing builder 3.2.3
remote:        Installing mini_portile2 2.1.0
remote:        Installing erubis 2.7.0
remote:        Installing rack 2.0.1
remote:        Installing nio4r 1.2.1 with native extensions
remote:        Installing websocket-extensions 0.1.2
remote:        Installing mime-types-data 3.2016.0521
remote:        Installing jsonapi-parser 0.1.1.beta3
remote:        Installing jsonapi-renderer 0.1.1.beta1
remote:        Installing arel 7.1.4
remote:        Installing bcrypt 3.1.11 with native extensions
remote:        Installing method_source 0.8.2
remote:        Installing pg 0.19.0 with native extensions
remote:        Installing puma 3.7.0 with native extensions
remote:        Installing rack-cors 0.4.0
remote:        Using bundler 1.13.7
remote:        Installing thor 0.19.4
remote:        Installing rails_serve_static_assets 0.0.5
remote:        Installing rails_stdout_logging 0.0.5
remote:        Installing tzinfo 1.2.2
remote:        Installing nokogiri 1.7.0.1 with native extensions
remote:        Installing rack-test 0.6.3
remote:        Installing sprockets 3.7.1
remote:        Installing websocket-driver 0.6.5 with native extensions
remote:        Installing mime-types 3.1
remote:        Installing jsonapi 0.1.1.beta6
remote:        Installing rails_12factor 0.0.3
remote:        Installing activesupport 5.0.1
remote:        Installing mail 2.6.4
remote:        Installing globalid 0.3.7
remote:        Installing activemodel 5.0.1
remote:        Installing case_transform 0.2
remote:        Installing activejob 5.0.1
remote:        Installing activerecord 5.0.1
remote:        Installing rails-dom-testing 2.0.2
remote:        Installing loofah 2.0.3
remote:        Installing rails-html-sanitizer 1.0.3
remote:        Installing actionview 5.0.1
remote:        Installing actionpack 5.0.1
remote:        Installing actionmailer 5.0.1
remote:        Installing actioncable 5.0.1
remote:        Installing active_model_serializers 0.10.4
remote:        Installing railties 5.0.1
remote:        Installing sprockets-rails 3.2.0
remote:        Installing rails 5.0.1
remote:        Bundle complete! 17 Gemfile dependencies, 50 gems now installed.
remote:        Gems in the groups development and test were not installed.
remote:        Bundled gems are installed into ./vendor/bundle.
remote:        Bundle completed (26.62s)
remote:        Cleaning up the bundler cache.
remote: -----> Detecting rake tasks
remote: 
remote: -----> Discovering process types
remote:        Procfile declares types     -> web
remote:        Default types for buildpack -> console, rake, worker
remote: 
remote: -----> Compressing...
remote:        Done: 26.3M
remote: -----> Launching...
remote:        Released v5
remote:        https://dry-hollows-40181.herokuapp.com/ deployed to Heroku
remote: 
remote: Verifying deploy... done.
To https://git.heroku.com/dry-hollows-40181.git
 * [new branch]      master -> master
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ 
```

**Comment by glasneed** [comment]
The name of the GIT remote is: 

```
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ git remote -v
heroku	https://git.heroku.com/dry-hollows-40181.git (fetch)
heroku	https://git.heroku.com/dry-hollows-40181.git (push)
origin	https://github.com/conorjennings/chores.git (fetch)
origin	https://github.com/conorjennings/chores.git (push)
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ 
```

**Comment by BenGitsCode** [comment]
When you say `git push heroku master` you are using `git` to `push` your `master` branch to the `heroku` remote. What did you rename your remote to?

**Comment by glasneed** [comment]
Attached is what the name is in the heroku configuration page. Also attached is my client app that I had set up to connect to  ``` https://kids-chores-nh.herokuapp.com/ ``` in config.js. I was able to sign in and set up a user but there is no chores table created when I look at the prod DB.

As for "_what did you rename your remote to_" I don't know what you mean. I don't recall renaming it unless it was in the instructions for Heroku?


And here is the terminal log of the DB migration:

```

conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ heroku run rake db:migrate
Running rake db:migrate on ⬢ dry-hollows-40181... up, run.9834 (Free)
   (22.3ms)  CREATE TABLE "schema_migrations" ("version" character varying PRIMARY KEY)
   (8.8ms)  CREATE TABLE "ar_internal_metadata" ("key" character varying PRIMARY KEY, "value" character varying, "created_at" timestamp NOT NULL, "updated_at" timestamp NOT NULL)
   (1.3ms)  SELECT pg_try_advisory_lock(497535716080682685);
  ActiveRecord::SchemaMigration Load (1.3ms)  SELECT "schema_migrations".* FROM "schema_migrations"
Migrating to CreateUsers (1)
   (1.2ms)  BEGIN
== 1 CreateUsers: migrating ===================================================
-- create_table(:users, {})
   (7.2ms)  CREATE TABLE "users" ("id" serial primary key, "email" character varying NOT NULL, "token" character varying NOT NULL, "password_digest" character varying NOT NULL, "created_at" timestamp NOT NULL, "updated_at" timestamp NOT NULL)
   (3.6ms)  CREATE UNIQUE INDEX  "index_users_on_email" ON "users"  ("email")
   (7.6ms)  CREATE UNIQUE INDEX  "index_users_on_token" ON "users"  ("token")
   -> 0.0277s
== 1 CreateUsers: migrated (0.0278s) ==========================================

  SQL (1.5ms)  INSERT INTO "schema_migrations" ("version") VALUES ($1) RETURNING "version"  [["version", "1"]]
   (3.8ms)  COMMIT
Migrating to CreateExamples (2)
   (1.1ms)  BEGIN
== 2 CreateExamples: migrating ================================================
-- create_table(:examples, {})
   (8.1ms)  CREATE TABLE "examples" ("id" serial primary key, "text" text NOT NULL, "user_id" integer NOT NULL, "created_at" timestamp NOT NULL, "updated_at" timestamp NOT NULL, CONSTRAINT "fk_rails_ae101d15cc"
FOREIGN KEY ("user_id")
  REFERENCES "users" ("id")
)
   (3.3ms)  CREATE  INDEX  "index_examples_on_user_id" ON "examples"  ("user_id")
   -> 0.0169s
== 2 CreateExamples: migrated (0.0169s) =======================================

  SQL (1.3ms)  INSERT INTO "schema_migrations" ("version") VALUES ($1) RETURNING "version"  [["version", "2"]]
   (3.0ms)  COMMIT
Migrating to CreateChores (20170427200847)
   (1.1ms)  BEGIN
== 20170427200847 CreateChores: migrating =====================================
-- create_table(:chores)
   (6.7ms)  CREATE TABLE "chores" ("id" serial primary key, "task" character varying, "priority" integer, "due_on" date, "created_at" timestamp NOT NULL, "updated_at" timestamp NOT NULL)
   -> 0.0074s
== 20170427200847 CreateChores: migrated (0.0075s) ============================

  SQL (1.3ms)  INSERT INTO "schema_migrations" ("version") VALUES ($1) RETURNING "version"  [["version", "20170427200847"]]
   (2.6ms)  COMMIT
Migrating to AddUserToChores (20170427204144)
   (1.1ms)  BEGIN
== 20170427204144 AddUserToChores: migrating ==================================
-- add_reference(:chores, :user, {:index=>true, :foreign_key=>true})
   (1.4ms)  ALTER TABLE "chores" ADD "user_id" integer
   (3.3ms)  CREATE  INDEX  "index_chores_on_user_id" ON "chores"  ("user_id")
   (2.5ms)  ALTER TABLE "chores" ADD CONSTRAINT "fk_rails_b880541aa1"
FOREIGN KEY ("user_id")
  REFERENCES "users" ("id")

   -> 0.0116s
== 20170427204144 AddUserToChores: migrated (0.0117s) =========================

  SQL (1.2ms)  INSERT INTO "schema_migrations" ("version") VALUES ($1) RETURNING "version"  [["version", "20170427204144"]]
   (2.4ms)  COMMIT
  ActiveRecord::InternalMetadata Load (1.2ms)  SELECT  "ar_internal_metadata".* FROM "ar_internal_metadata" WHERE "ar_internal_metadata"."key" = $1 LIMIT $2  [["key", :environment], ["LIMIT", 1]]
   (1.1ms)  BEGIN
  SQL (1.3ms)  INSERT INTO "ar_internal_metadata" ("key", "value", "created_at", "updated_at") VALUES ($1, $2, $3, $4) RETURNING "key"  [["key", "environment"], ["value", "production"], ["created_at", 2017-05-03 00:42:16 UTC], ["updated_at", 2017-05-03 00:42:16 UTC]]
   (2.6ms)  COMMIT
   (1.2ms)  SELECT pg_advisory_unlock(497535716080682685)
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ 


```
![screen shot 2017-05-02 at 9 09 23 pm](https://cloud.githubusercontent.com/assets/26227953/25645465/526495f6-2f7c-11e7-9026-dae28c91263d.png)

**Comment by BenGitsCode** [comment]
What is the output of `heroku info`?

**Comment by glasneed** [comment]
```
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ heroku info
=== dry-hollows-40181
Addons:         heroku-postgresql:hobby-dev
Auto Cert Mgmt: false
Dynos:          web: 1
Git URL:        https://git.heroku.com/dry-hollows-40181.git
Owner:          conorsmail@gmail.com
Region:         us
Repo Size:      70 KB
Slug Size:      26 MB
Stack:          cedar-14
Web URL:        https://dry-hollows-40181.herokuapp.com/
conor@LIBP45P-10984DY ~/wdi/projects/chores-server (master) $ 

```

**Comment by glasneed** [comment]
I checked that heroku config dashboard and now there are more apps since I set it up with the one app last week (that kids chores NH). See attached...
![screen shot 2017-05-02 at 9 23 01 pm](https://cloud.githubusercontent.com/assets/26227953/25645608/b0fb7430-2f7d-11e7-85ec-1d0f2f493afe.png)

**Comment by BenGitsCode** [comment]
It looks like you've renamed, and run heroku create a few times? That will create several apps, I thought you were renaming the same one. 

I would delete them all and start with a clean deployment to isolate the issue.

First, ensure that the code responsible for what was missing, and works locally, is on both your local and github remote`master` branches.

Then try removing the remote locally. Then follow the deployment steps again and create a new app and try pushing to that.

**Comment by glasneed** [comment]
OK thanks. I'm not going to start all over tonight as I've spent 2 hours on this now and gotten nowhere.

 I'll do this in person with someone this week as this just does not work via the issue log. I simply followed the steps in the Heroku guide last week (my first time using this heroku app) and then went to do a deploy tonight as I explained and it failed. I don't have the necessary depth of knowledge on how do remove local and remote branches nor am I going to try tonight.

Thank you for your help this evening.
